main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -384,7 +384,29 @@ object Verifier {
           check(expected = MonoType.Region)(actual = t2, loc)
           check(expected = MonoType.Unit)(actual = tpe, loc)
 
-        case _ => tpe // TODO: VERIFIER: Add rest
+        case AtomicOp.GetField(field) =>
+          checkEq(tpe, javaClassToMonoType(field.getType), loc)
```

Comment:
I think, now that you've gone into checking the java types, that you should try verifying the other types too, like checking that `List(t) = ts; checkJavaEq(field.getDeclaringClass, t, ...)`

Reply:
But in this case, `tpe` may be a subclass of the declaring class, and only the field itself is given.
Should I walk the subtree of `getDeclaringClass` looking for a match or just ignore it? Or maybe I missed something.
Specifically fails here: main/test/flix/Test.Exp.Jvm.PutField.flix:87:9

Reply:
new commit addresses the problem, but not super elegantly